infra/storage/file_transfer_managers/s3Mgr.class.php

```diff
@@ -334,3 +374,41 @@ public function registerStreamWrapper()
 		$this->s3->registerStreamWrapper();
 	}
 }
+
+class RefreshableRole extends AbstractRefreshableCredentials
```

Comment:
Consider moving to another file , next to the s3Mgr

Reply:
Don't see it being used elsewhere and I already have al the requirements loaded asa part of this class. I will later need it in another place we can move it to a separate file.